packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3321,7 +3373,14 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   /// The element will still have a valid parent when this is called. After this
   /// is called, [deactivateChild] is called to sever the link to this object.
   @protected
-  void forgetChild(Element child);
+  @mustCallSuper
```

Comment:
This is to deal with the case where the early rebuilt element gets into shenanigan.
for example child1(key1)  and child2(key2).
child2 gets early rebuilt and reserved key2 for the slot of child2
then the swap and layoutbuilder rebuild
it will first build the slot of original child1 with child2(key2). This will cause child2 gets retake from the original slot of child2. Which will call the forgetChild of that slot and clear any trace of child2 from that slot. while it build the original slot of child2, there is no way to remove the reservation from child2 because there is no any evidence left in that slot.

I feel this is a rather dirty breaking change. Now sure if there is any way around it

Reply:
i don't think it's that dirty. it seems reasonable that the base class have an implementation here and that we'd required subclasses to call it.